app/controllers/settings_controller.rb

```diff
@@ -48,6 +48,8 @@ def edit
         if value.is_a?(Array)
           # remove blank values in array settings
           value.delete_if(&:blank?)
+        elsif value.is_a?(Hash)
+          value.delete_if {|k,v| v.blank? }
```

Comment:
Unused block argument - `k`. If it's necessary, use `_` or `_k` as an argument name to indicate that it won't be used.<br>Space missing after comma.<br>Space between { and | missing.